modules/transformer-kinesis/src/main/scala/com/snowplowanalytics/snowplow/rdbloader/transformer/kinesis/parquet/ParquetSink.scala

```diff
@@ -42,14 +44,12 @@ object ParquetSink {
                                                            (path: SinkPath): Pipe[F, Transformed.Data, Unit] = {
     transformedData =>
 
-      val targetPath = uri.toString + window.getDir + "/" + path.value
+      val targetPath = Path.of(uri.toString, window.getDir, path.value)
       val schemaCreation = createSchemaFromTypes(resources, window).value
 
       Stream.eval(schemaCreation).flatMap {
         case Left(error) =>
-          println(error) //TODO
-          Stream.emit(())
-
+          Stream.raiseError[F](new RuntimeException(s"Error while building parquet schema: $error"))
```

Comment:
RuntimeException has [a constructor that lets you set the `cause`](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/RuntimeException.html#%3Cinit%3E(java.lang.String,java.lang.Throwable))

```
RuntimeException(String message, Throwable cause)
```

Reply:
But we don't have a `Throwable` here which could be set as cause...right?

Reply:
Ah, I understand now -- it's a `LoaderIgluError` not a Throwable.  My mistake.

The problem here is that we should not ever stringify a `LoaderIgluError`, because this exception message will end up in the logs.  If you look [the definition of `LoaderIgluError`](https://github.com/snowplow/snowplow-badrows/blob/2.1.1/src/main/scala/com.snowplowanalytics.snowplow.badrows/FailureDetails.scala#L170-L182) you see it might contain user-sent data in the `value` fields.  These days we have a rule that we never log user-sent data.  It's for two reasons:

- Access to logs is harder to manage, if it possibly contains any personally identifiable information.
- Log4j made us all scared of printing messages if we're not in control of them!